modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/reader/ZipDumpReader.java

```diff
@@ -39,14 +46,56 @@ public static ZipDumpReader create( SystemLoadListener listener, final Path base
             final SeekableByteChannel seekableByteChannel =
                 Files.newByteChannel( basePath.resolve( dumpName + ".zip" ), EnumSet.of( StandardOpenOption.READ ) );
             final ZipFile zipFile = new ZipFile( seekableByteChannel );
-            return new ZipDumpReader( listener, PathRef.of( dumpName ), zipFile );
+
+            return create( listener, dumpName, zipFile );
         }
         catch ( IOException e )
         {
             throw new UncheckedIOException( e );
         }
     }
 
+    private static ZipDumpReader create( final SystemLoadListener listener, final String dumpName, final ZipFile zipFile )
+    {
+        if ( zipFile.getEntry( "dump.json" ) != null )
+        {
+            return new ZipDumpReader( listener, PathRef.of(), zipFile );
+        }
+        else if ( zipFile.getEntry( dumpName + "/dump.json" ) != null )
+        {
+            return new ZipDumpReader( listener, PathRef.of( dumpName ), zipFile );
+        }
+        else
+        {
+            String rootDumpDir = null;
+
+            final Enumeration<ZipArchiveEntry> entries = zipFile.getEntries();
+
+            while ( entries.hasMoreElements() )
+            {
+                final ZipArchiveEntry entry = entries.nextElement();
+
+                final Matcher matcher = ROOT_DUMP_DIR_PATTERN.matcher( entry.getName() );
+
+                if ( matcher.find() )
```

Comment:
I would prefer `matches()` for clarity.

Reply:
@rymsha In this case I prefer matcher, because pattern already compiled and in order to get a name of `rootDir` will be enough to call `matcher.group( 1 )`

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Matcher.html#matches()

Reply:
Ah, I thought you meant https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#matches(java.lang.String)